src/collision/cm_trace.c

```diff
@@ -55,81 +60,45 @@ static __thread struct {
 	 */
 	vec3_t offsets[8];
 
-	/**
-	 * @brief The contents mask to collide with, as provided by the user.
-	 */
-	int32_t contents;
-
-	/**
-	 * @brief The transformation matrix for plane collisions, as provided by the user.
-	 */
-	mat4_t matrix;
-
-	/**
-	 * @brief The transformation matrix for start/end/bounds, for the node tests, as provided by the user.
-	 */
-	mat4_t inverse_matrix;
-
-	/**
-	 * @brief True if matrix is not the identity.
-	 */
-	_Bool is_transformed;
-
 	/**
 	 * @brief The brush cache, to avoid multiple tests against the same brush.
 	 */
 	int32_t brush_cache[128];
 
 	/**
-	 * @brief The trace result.
+	 * @brief The trace fraction not taking any epsilon nudging into account.
 	 */
-	cm_trace_t trace;
+	float unnudged_fraction;
 
 	/**
-	 * @brief The trace fraction not taking any epsilon nudging into account.
+	 * @brief The trace result.
 	 */
-	float unnudged_fraction;
-} data;
+	cm_trace_t trace;
+} cm_trace_data_t;
 
 /**
  * @brief
  */
-static inline _Bool Cm_BrushAlreadyTested(int32_t brush_num) {
-	const int32_t hash = brush_num & (lengthof(data.brush_cache) - 1);
+static inline _Bool Cm_BrushAlreadyTested(cm_trace_data_t *data, int32_t brush_num) {
+	const int32_t hash = brush_num & (lengthof(data->brush_cache) - 1);
 
-	const _Bool skip = (data.brush_cache[hash] == brush_num);
+	const _Bool skip = (data->brush_cache[hash] == brush_num);
 
-	data.brush_cache[hash] = brush_num;
+	data->brush_cache[hash] = brush_num;
 
 	return skip;
 }
 
-/**
- * @brief 
- */
-static inline _Bool Cm_TraceIntersect(const cm_bsp_brush_t *brush) {
```

Comment:
Because transforms are done before / after the trace, we never have to adjust brush bounds. So this is just `Box3_Intersects(data->abs_bounds, brush->bounds)` now.